extensions/control-plane/provision/provision-bigquery/src/main/java/org/eclipse/edc/connector/provision/gcp/BigQueryProvisioner.java

```diff
@@ -78,11 +74,10 @@ public CompletableFuture<StatusResult<ProvisionResponse>> provision(
         var resourceName = tableName + "-table";
 
         // TODO update target with the generated table name.
-
         try {
-            GcpServiceAccount serviceAccount = null;
-            if (serviceAccountName != null) {
-                serviceAccount = iamService.getServiceAccount(serviceAccountName);
+            var serviceAccountName = resourceDefinition.getServiceAccountName();
+            if (serviceAccountName == null) {
+                serviceAccountName = gcpConfiguration.serviceAccountName();
```

Comment:
this logic is still duplicated (here and in `GcsProvisioner`), we could move this if in the `iamService` and let the `BigQueryFactory` accept a `GcpServiceAccount`, that will always be non-null

Reply:
In the IamService ctor we pass the gcpConfiguration, and then we can handle the logic inside: I like it!

Reply:
Done